shell/common/shell.cc

```diff
@@ -740,6 +740,21 @@ void Shell::OnEngineHandlePlatformMessage(
       });
 }
 
+// |shell::Engine::Delegate|
+void Shell::OnEngineRestart() {
```

Comment:
This will deadlock on configurations where the task runners are the same (test/embedder configurations). Please use `fml::TaskRunner::RunNowOrPostTask` instead.

Reply:
Please add the assertions for setup and the thread checker as done in all the other methods in this file.

Reply:
Wow good catch on the threading issue! thanks!

Reply:
Added the same assertions, though I believe we don't really care to assert that we're on the ui thread